### DISCUSSION TITLE
Is there a config.py API to spawn an external command or userscript?

### DISCUSSION THREAD

**Discussion by emanuele6** [qutebrowser/qutebrowser#5803] [Q&A] [answered]
and is there an API to run a qutebrowser command in general from `config.py`?

**Comment by The-Compiler** [comment, marked as answer]
There's no official API for anything yet (other than setting config options). There are a couple of unofficial things in the `qutebrowser.api` namespace (such as registering request interceptors or commands) but even those might still change before being an official extension API (see #30 for details).

Can you elaborate on what you're trying to do exactly?

**Reply by emanuele6** [reply]
I saw [this post on r/qutebrowser](https://www.reddit.com/r/qutebrowser/comments/jbjnrg/timed_loops/) and I thought that spawning non-detached the following userscript:
```sh
#!/bin/sh
while sleep 60; do
    notify-send 'qutebrowser is running'
    : do something
    echo ':SOME_QB_COMMAND SOME_ARG' > "$QUTE_FIFO"
done
```
..when `qutebrowser` loads `config.py` would be pretty convenient since this script would die when `qutebrowser` is closed.

There are [other solutions](https://www.reddit.com/r/qutebrowser/comments/jbjnrg/timed_loops/g8vsbuc?utm_source=share&utm_medium=web2x&context=3) of course and I don't know for what reasons [u/fimari](https://www.reddit.com/user/fimari/) needs this.

I mainly asked just because I was wondering if there was an API for such things (running commands, spawning).

**Reply by The-Compiler** [reply]
Nothing of that is part of `qutebrowser.api` so far. You can run external commands via Python's normal `subprocess` module (or Qt's `QProcess`) - at some point in the future, it might make sense to expose the `GUIProcess` class qutebrowser uses for `:spawn` (so you e.g. get a message about its exit status).

However, in general, `config.py` is intended for configuration and not extensions-like hacks, so strictly speaking there's no API for anything at all :wink: (other than the `config` object).
